config/config-api/src/com/thoughtworks/go/config/ValidationContext.java

```diff
@@ -110,6 +110,21 @@ public JobConfig getJob() {
         } else if (immediateParent.getClass().equals(klass)) {
             return (T) immediateParent;
         }
+        else
+        {
+            // added because of higher hierarchy of configuration types.
```

Comment:
No, it won't. It is still needed by MergeEnvironmentConfig and MergePipelineConfigs for the same reason